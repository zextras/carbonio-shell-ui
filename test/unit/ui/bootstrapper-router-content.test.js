/*
 * *** BEGIN LICENSE BLOCK *****
 * Copyright (C) 2011-2020 ZeXtras
 *
 * The contents of this file are subject to the ZeXtras EULA;
 * you may not use this file except in compliance with the EULA.
 * You may obtain a copy of the EULA at
 * http://www.zextras.com/zextras-eula.html
 * *** END LICENSE BLOCK *****
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

jest.mock('../../../src/bootstrap/bootstrapper-lazy-loader');

import BootstrapperRouterContent from '../../../src/bootstrap/bootstrapper-router-content';

describe('Shell Router', () => {

	test('Route to Shell with no Account', () => {
		const router = renderer.create(
			<MemoryRouter>
				<BootstrapperRouterContent accounts={[]} />
			</MemoryRouter>
		);
		expect(router.toJSON()).toMatchSnapshot();
	});

	test('Route to Shell with Account', () => {
		const router = renderer.create(
			<MemoryRouter>
				<BootstrapperRouterContent accounts={[{}]} />
			</MemoryRouter>
		);
		expect(router.toJSON()).toMatchSnapshot();
	});

	test('Route to Logout with Account', () => {
		const router = renderer.create(
			<MemoryRouter initialEntries={['/logout']}>
				<BootstrapperRouterContent accounts={[{}]} />
			</MemoryRouter>
		);
		expect(router.toJSON()).toMatchSnapshot();
	});

});
