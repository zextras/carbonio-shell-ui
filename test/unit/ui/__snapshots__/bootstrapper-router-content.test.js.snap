// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shell Router Route to Logout with Account 1`] = `"LazyLogoutView"`;

exports[`Shell Router Route to Shell with Account 1`] = `"LazyShellView"`;

exports[`Shell Router Route to Shell with no Account 1`] = `"LazyLoginView"`;
